src/connection.cpp

```diff
@@ -132,15 +119,18 @@ std::vector<std::string> Connection::fetchInterfaces()
         if (std::strcmp(ifa->ifa_name, "lo") == 0)
             continue;
 
-        interfaces.emplace_back(ifa->ifa_name);
-    }
+        bool duplicate = false;
+        for (const auto& name : interfaces) {
+            if (std::strcmp(ifa->ifa_name, name.c_str()) == 0) {
+                duplicate = true;
+                break;
+            }
+        }
 
-    // TODO: It must have a better way to remove duplicates instead of creating
-    //       a set to filter out them. Perhaps changing the return type?
```

Comment:
Did we get anything good considering this TODO?

Reply:
I changed the way duplicate validation is done / without set... Only this

Reply:
> I changed the way duplicate validation is done / without set... Only this

Yes but using set already do it for us and also put it in alphabetic order right? There's any advantage?

Reply:
I don't think so. Just better to read (at least in my pov)

Reply:
Did you mind using set instead?